pilot/helpers/agents/Developer.py

```diff
@@ -365,6 +433,10 @@ def execute_task(self, convo, task_description, task_steps, test_command=None, r
         convo.save_branch(function_uuid)
 
         for (i, step) in enumerate(task_steps):
+            # This means we are still loading the project
+            if self.project.last_iteration is not None:
```

Comment:
Should we use the same check (like `if self.skip_dev_steps`)? everywhere so we don't end up in some weird case where one part of the code thinks we've finished loading and the other still thinks we're loading?